tests/integration/targets/latest/tasks/controllers.yml

```diff
@@ -0,0 +1,84 @@
+---
+- name: "CONTROLLER 1: Necessary info creation"
+  networktocode.nautobot.controller:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: Test Controller One
+    state: present
+    status: "Active"
+  register: test_one_controller
+
+- name: "CONTROLLER 1: ASSERT - Necessary info creation"
+  assert:
+    that:
+      - test_one_controller is changed
+      - test_one_controller['diff']['before']['state'] == "absent"
+      - test_one_controller['diff']['after']['state'] == "present"
+      - test_one_controller['controller']['name'] == "Test Controller One"
+      - test_one_controller['controller']['object_type'] == "dcim.controller"
+      - test_one_controller['msg'] == "controller Test Controller One created"
+
+- name: "CONTROLLER 2: Create duplicate"
+  networktocode.nautobot.controller:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: Test Controller One
+    state: present
+    status: "Active"
+  register: test_two
+
+- name: "CONTROLLER 2: ASSERT - Create duplicate"
+  assert:
+    that:
+      - not test_two['changed']
+      - test_two['controller']['name'] == "Test Controller One"
+      - test_two['msg'] == "controller Test Controller One already exists"
+
```

Comment:
Can you create a test that creates a controller using as many fields on the module as you can?